### PR TITLE
s3 quick start

### PR DESCRIPTION
**Pull request by y-f-u**

### CODE REVIEW

File: s3/README.md [hunk 1]
```diff
@@ -0,0 +1,207 @@
+# Spice.ai Quickstart Tutorials using S3
+
+## For public S3 bucket
+
+**Step 1.** If you haven't already initialized a new project, you need to do so. Then, start the Spice Runtime.
```
Comment (phillipleblanc) [inline s3/README.md]:
```suggestion
**Step 1.** Initialize a Spice application, then start the Spice runtime.
```

File: s3/README.md [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+```bash
+spice init s3-demo-project
+```
+
+```bash
+cd s3-demo-project
+spice run
+```
+
+**Step 2.** We now configure the dataset from S3:
```
Comment (phillipleblanc) [inline s3/README.md]:
```suggestion
**Step 2.** Configure the dataset to connect to S3:
```

File: s3/README.md [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+Select "y" when prompted whether to locally accelerate the dataset:
+
+```bash
+Locally accelerate (y/n)? y
+```
+
+We should now see the following output:
+
+```
+Dataset settings written to `datasets/taxi_trips/dataset.yaml`!
```
Comment (phillipleblanc) [inline s3/README.md:49]: Would be good to show the contents of `dataset.yaml` for completeness

File: s3/README.md [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+
+**LIMITATION:** Spice.ai currently does not support discovering nested directories in S3. Please ensure that the parquet files are in the root of the specified S3 path.
+
+Select "y" when prompted whether to locally accelerate the dataset:
+
+```bash
+Locally accelerate (y/n)? y
+```
+
+We should now see the following output:
```
Comment (phillipleblanc) [inline s3/README.md]:
```suggestion
The following output is shown:
```

File: s3/README.md [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+
+```
+Dataset settings written to `datasets/taxi_trips/dataset.yaml`!
+```
+
+You also should see the following in the Spice runtime terminal :
```
Comment (phillipleblanc) [inline s3/README.md]:
```suggestion
The following output is shown in the Spice runtime terminal:
```

File: s3/README.md [hunk 6]
```diff
@@ -0,0 +1,207 @@
...
+
+**Step 3.** Run queries against the dataset using the Spice SQL REPL.
+
+In a new terminal, start the Spice SQL REPL
+
+```bash
+spice sql
+```
+
+Check the taxi_trips table exists:
```
Comment (phillipleblanc) [inline s3/README.md]:
```suggestion
Check that the taxi_trips table exists:
```

File: s3/README.md [hunk 7]
```diff
@@ -0,0 +1,207 @@
...
++---------------+--------------------+-------------+------------+
+| datafusion    | public             | taxi_trips  | BASE TABLE |
+| datafusion    | information_schema | tables      | VIEW       |
+| datafusion    | information_schema | views       | VIEW       |
+| datafusion    | information_schema | columns     | VIEW       |
+| datafusion    | information_schema | df_settings | VIEW       |
++---------------+--------------------+-------------+------------+
+```
+
+You can now now query `taxi_trips` in the runtime.
```
Comment (phillipleblanc) [inline s3/README.md]:
```suggestion
Query against the `taxi_trips` table in the runtime.
```

File: s3/README.md [hunk 8]
```diff
@@ -0,0 +1,207 @@
...
+  ```
+- Create access key contains `aws_access_key_id` and `aws_secret_access_key` for the IAM user under `Security credentials` tab.
+
+**Step 3.** Log into S3 using the Spice CLI.
+
+```
+spice login s3 -k <aws_access_key_id> -s <aws_secret_access_key>
+```
+
+You should see the following output:
```
Comment (phillipleblanc) [inline s3/README.md]:
```suggestion
The following output is shown:
```

File: s3/README.md [hunk 9]
```diff
@@ -0,0 +1,207 @@
...
+      },
+      {
+        "Effect": "Allow",
+        "Action": ["s3:GetObject"],
+        "resource": "arn:aws:s3:::yourcompany-bucketname-datasets/*"
+      }
+    ]
+  }
+  ```
+- Create access key contains `aws_access_key_id` and `aws_secret_access_key` for the IAM user under `Security credentials` tab.
```
Comment (phillipleblanc) [inline s3/README.md]:
```suggestion
- Create an access key that contains `aws_access_key_id` and `aws_secret_access_key` for the IAM user under `Security credentials` tab.
```

Comment (phillipleblanc) [inline s3/README.md]: Let's have a screenshot here too of what they need to click

File: s3/README.md [hunk 10]
```diff
@@ -0,0 +1,207 @@
...
+spice login s3 -k <aws_access_key_id> -s <aws_secret_access_key>
+```
+
+You should see the following output:
+
+```
+Successfully logged in to s3
+```
+
+**Step 4.** If you haven't already initialized a new project, you need to do so. Then, start the Spice Runtime.
```
Comment (phillipleblanc) [inline s3/README.md]:
```suggestion
**Step 4.** Initialize a Spice application, then start the Spice runtime.
```

File: s3/README.md [hunk 11]
```diff
@@ -0,0 +1,207 @@
...
+```bash
+spice init s3-demo-project
+```
+
+```bash
+cd s3-demo-project
+spice run
+```
+
+**Step 5.** We now configure the dataset from S3:
```
Comment (phillipleblanc) [inline s3/README.md]:
```suggestion
**Step 5.** Configure the dataset to connect to S3:
```

File: s3/README.md [hunk 12]
```diff
@@ -0,0 +1,207 @@
...
+from: s3://yourcompany-bucketname-datasets/taxi_trips/
+```
+
+Select "y" when prompted whether to locally accelerate the dataset:
+
+```bash
+Locally accelerate (y/n)? y
+```
+
+We should now see the following output:
```
Comment (phillipleblanc) [inline s3/README.md]:
```suggestion
The following output is shown:
```

File: s3/README.md [hunk 13]
```diff
@@ -0,0 +1,207 @@
...
+
+```
+Dataset settings written to `datasets/taxi_trips/dataset.yaml`!
+```
+
+If the login credentials were entered correctly, your dataset will have loaded into the runtime. You should see the following in the Spice runtime terminal :
```
Comment (phillipleblanc) [inline s3/README.md]:
```suggestion
If the login credentials were entered correctly, the dataset will have loaded into the runtime. The following output is shown in the Spice runtime terminal:
```

File: s3/README.md [hunk 14]
```diff
@@ -0,0 +1,207 @@
...
+```
+
+If the login credentials were entered correctly, the dataset will have loaded into the runtime. The following output is shown in the Spice runtime terminal:
+
+```
+2024-03-26T22:18:02.062394Z  INFO runtime: Loaded dataset: taxi_trips
+2024-03-26T22:18:02.062462Z  INFO runtime::dataconnector: Refreshing data for taxi_trips
+```
+
+**Step 6.** Run queries against the dataset using the Spice SQL REPL. (see Step 3 above)
```
Comment (digadeesh) [inline s3/README.md]: Just put the SQL here instead of referring to step 3 above.  There are actually 2 step 3's above :).

Comment (y-f-u) [reply s3/README.md]: god, my math teacher would be mad at me.

**Review by phillipleblanc — DISMISSED** [review]

File: s3/README.md [hunk 15]
```diff
@@ -0,0 +1,225 @@
...
+description: taxi trips in s3
+```
+
+Specify the location of the dataset:
+
+```bash
+from: s3://spiceai-demo-datasets/taxi_trips/2024/
+```
+
+**LIMITATION:** Spice.ai currently does not support discovering nested directories in S3. Please ensure that the parquet files are in the root of the specified S3 path.
```
Comment (lukekim) [inline s3/README.md]: I don't think we need limitations in quickstarts, I would just add this to the S3 section of docs.

File: s3/README.md [hunk 16]
```diff
@@ -0,0 +1,225 @@
...
+| 18.45                        | 3.05                       | 1               | 9               |
+| 25.811736633324347           | 1.545956750046378          | 140162          |                 |
++------------------------------+----------------------------+-----------------+-----------------+
+
+Query took: 0.015628708 seconds
+```
+
+## For private S3 bucket
+
+**LIMITATION:** Spice.ai currently only supports authentication with AWS S3 using `aws_access_key_id` and `aws_secret_access_key`. Please [submit an issue](https://github.com/spiceai/spiceai/issues/new?template=feature_request.md) if you would like to see support for other authentication methods.
```
Comment (lukekim) [inline s3/README.md]: Same comment.

**Review by phillipleblanc — APPROVED** [review]

**Pull request merged**